src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java

```diff
@@ -0,0 +1,179 @@
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Implementation of a check to verify if specific files exist in their correct
+ * folders locations.
+ *
+ * @author CheckStyle Team
+ */
+public class RegexpOnFilenameCheck extends AbstractFileSetCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MATCH = "regexp.filepath.match";
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISMATCH = "regexp.filepath.mismatch";
+
+    /** Compiled regexp to match a folder. */
+    private Pattern folderPattern;
+    /** Compiled regexp to match a file. */
+    private Pattern fileNamePattern;
+    /** Whether to look for a file name match or mismatch. */
+    private boolean match = true;
+    /** Whether to ignore the file's extension when looking for matches. */
+    private boolean ignoreFileExtensions;
+
+    /**
+     * Setter for folder format.
+     *
+     * @param folderPattern format of folder.
+     * @throws org.apache.commons.beanutils.ConversionException if unable to
+     *         create Pattern object.
+     */
+    public void setFolderPattern(String folderPattern) {
+        this.folderPattern = CommonUtils.createPattern(folderPattern);
+    }
+
+    /**
+     * Setter for file name format.
+     *
+     * @param fileNamePattern format of file.
+     * @throws org.apache.commons.beanutils.ConversionException if unable to
+     *         create Pattern object.
+     */
+    public void setFileNamePattern(String fileNamePattern) {
+        this.fileNamePattern = CommonUtils.createPattern(fileNamePattern);
+    }
+
+    /**
+     * Sets whether the check should look for a file name match or mismatch.
+     *
+     * @param match check's option for matching file names.
+     */
+    public void setMatch(boolean match) {
+        this.match = match;
+    }
+
+    /**
+     * Sets whether file name matching should drop the file extension or not.
+     *
+     * @param ignoreFileExtensions check's option for ignoring file extension.
+     */
+    public void setIgnoreFileExtensions(boolean ignoreFileExtensions) {
+        this.ignoreFileExtensions = ignoreFileExtensions;
+    }
+
+    @Override
+    public void init() {
+        if (fileNamePattern == null && folderPattern == null) {
+            fileNamePattern = CommonUtils.createPattern("\\s");
+            match = true;
+        }
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+        String filePath;
+        String folderPath;
+
+        try {
+            filePath = file.getCanonicalPath();
+            folderPath = file.getParentFile().getCanonicalPath();
+        }
+        catch (IOException ex) {
+            filePath = file.getAbsolutePath();
```

Comment:
@mkordas 
`RegexpOnFilenameCheckTest.testException` showcases when this happens.
Truthfully, I don't think this exception can be reached normally since we have already read the file by this point. The main reason for the catch is because we can't throw the IOException as is, and like I said I just used the original implementation's logic for that.